clients/drcachesim/reader/config_reader.cpp

```diff
@@ -228,21 +228,19 @@ config_reader_t::configure_cache(cache_params_t &cache)
                 ERRMSG("Unusable cache size %s\n", size_str.c_str());
                 return false;
             }
-            if (cache.size <= 0 || !IS_POWER_OF_2(cache.size)) {
-                ERRMSG("Cache size (%llu) must be >0 and a power of 2\n",
-                       (unsigned long long)cache.size);
+            if (cache.size <= 0) {
+                ERRMSG("Cache size (%llu) must be >0\n", (unsigned long long)cache.size);
                 return false;
             }
         } else if (param == "assoc") {
-            // Cache associativity_. Must be a power of 2.
+            // Cache associativity_.
             if (!(*fin_ >> cache.assoc)) {
                 ERRMSG("Error reading cache assoc from "
                        "the configuration file\n");
                 return false;
             }
-            if (cache.assoc <= 0 || !IS_POWER_OF_2(cache.assoc)) {
-                ERRMSG("Cache associativity (%u) must be >0 and a power of 2\n",
-                       cache.assoc);
+            if (cache.assoc <= 0) {
```

Comment:
Right, I don't think the configuration requires a specific order for size and associativity settings, so it's a bit tricky to check here.  I also tend to think it's better for the cache itself to decide what configurations it will support, and the config reader should mostly check for gross errors.